bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageFactoryCreator.java

```diff
@@ -9,19 +9,43 @@
  */
 package io.pravega.storage.extendeds3;
 
+import com.google.common.base.Preconditions;
 import io.pravega.segmentstore.storage.ConfigSetup;
 import io.pravega.segmentstore.storage.StorageFactory;
 import io.pravega.segmentstore.storage.StorageFactoryCreator;
+import io.pravega.segmentstore.storage.StorageFactoryInfo;
+import io.pravega.segmentstore.storage.StorageManagerLayoutType;
+import io.pravega.segmentstore.storage.StorageManagerType;
+
 import java.util.concurrent.ScheduledExecutorService;
 
 public class ExtendedS3StorageFactoryCreator implements StorageFactoryCreator {
     @Override
-    public StorageFactory createFactory(ConfigSetup setup, ScheduledExecutorService executor) {
-        return new ExtendedS3StorageFactory(setup.getConfig(ExtendedS3StorageConfig::builder), executor);
+    public StorageFactory createFactory(StorageFactoryInfo storageFactoryInfo, ConfigSetup setup, ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(storageFactoryInfo, "storageFactoryInfo");
```

Comment:
TIP: You can annotate each field with `@NonNull`; Lombok will auto-generate the precondition checks.

Reply:
Same in the other classes.

Reply:
Fixed.